src/host.ts

```diff
@@ -85,3 +89,24 @@ export const getMostAvailableHost = async (
 
   return stickyHint
 }
+
+export const getSavedOrMostAvailableHost = async (
+  appId: string,
+  builder: Builder,
+  nHosts: number,
+  timeout: number
+): Promise<string> => {
+  if (hasStickyHost(appId)) {
+    log.debug(`Found sticky host saved locally`)
+    const {stickyHost, lastUpdated} = getStickyHost(appId)
+    const timeElapsed = moment.duration(moment().diff(lastUpdated))
+    if (timeElapsed.asHours() <= TTL_SAVED_HOST_HOURS) {
+      return stickyHost
+    }
+    log.debug(`Saved sticky host has expired`)
+  }
+  log.debug(`Finding a new sticky host`)
+  const newStickyHost = await getMostAvailableHost(appId, builder, nHosts, timeout)
+  saveStickyHost(appId, newStickyHost)
```

Comment:
You should check if the `newStickyHost` is `undefined` before saving it. There are cases where the `availability` requests might fail (timeout).